logbook-core/src/test/java/org/zalando/logbook/HeaderFiltersTest.java

```diff
@@ -99,14 +140,42 @@ void shouldRemoveHeaderByValue() {
     @Test
     void shouldRemoveAndChangeHeader() {
         final HeaderFilter unit = HeaderFilter.merge(
-                removeHeaders((key, value) -> "name".equals(key) && "Bob".equals(value)),
-                eachHeader((name, value) -> "name".equals(name) && "Alice".equals(value) ? "Carol" : value));
+                removeHeaders((key, value) ->
+                        "name".equals(key) && "Bob".equals(value)),
+                eachHeader((name, value) ->
+                        "name".equals(name) && "Alice".equals(value) ? "Carol" : value));
 
-        final Map<String, List<String>> filtered = unit.filter(MockHeaders.of("name", "Alice", "name", "Bob"));
+        final Map<String, List<String>> filtered = unit.filter(
+                HttpHeaders.empty()
+                        .update("name", "Alice", "Bob"));
 
         assertThat(filtered, not(hasEntry("name", singletonList("Alice"))));
         assertThat(filtered, not(hasEntry("name", singletonList("Bob"))));
         assertThat(filtered, hasEntry("name", singletonList("Carol")));
     }
 
+    @Test
+    void removesWholeHeader() {
+        final HeaderFilter unit = removeHeaders((name, value) ->
+                name.equals("Set-Cookie"));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/whiskeysierra/logbook/pullRequest?prid=5001975)